src/compiler/crystal/command.cr

```diff
@@ -563,7 +564,7 @@ class Crystal::Command
 
   private def error(msg, exit_code = 1)
     # This is for the case where the main command is wrong
-    @color = false if ARGV.includes?("--no-color")
+    @color = false if ARGV.includes?("--no-color") || ENV["TERM"]? == "dumb"
```

Comment:
This line IMO should be in general refactored due to a hacky way of (re)setting `@color` to a proper value - which should be done in line 524 already.

Reply:
Will line 524 run "in the case where the main command is wrong?".. If so the line could be deleted. But if not it can't. If it would run, why was this line added with this comment?

Reply:
Will line 524 run "in the case where the main command is wrong?".. If so the line could be deleted. But if not it can't. If it would run, why was this line added with this comment?